pilot/pkg/xds/endpoint_builder.go

```diff
@@ -64,6 +61,24 @@ func NewEndpointBuilder(clusterName string, proxy *model.Proxy, push *model.Push
 	}
 }
 
+func (b EndpointBuilder) DestinationRule() *networkingapi.DestinationRule {
```

Comment:
Usually we use pointer - not sure how EndpointBuilder is passed around, but we have been hit in the past by pass-by-value.